src/sentry/releasehealth/metrics.py

```diff
@@ -150,3 +158,185 @@ def _compute_crash_free_rate(data: Dict[str, float]) -> Optional[float]:
         crash_free_rate = 100 * max(0.0, crash_free_rate)
 
         return crash_free_rate
+
+    def get_release_adoption(
+        self,
+        project_releases: Sequence[Tuple[ProjectId, ReleaseName]],
+        environments: Optional[Sequence[EnvironmentName]] = None,
+        now: Optional[datetime] = None,
+        org_id: Optional[OrganizationId] = None,
+    ) -> ReleaseHealthBackend.ReleasesAdoption:
+        project_ids = list({x[0] for x in project_releases})
+        if org_id is None:
+            org_id = self._get_org_id(project_ids)
+
+        if now is None:
+            now = datetime.now(pytz.utc)
+
+        return self._get_release_adoption_impl(
+            now, org_id, project_releases, project_ids, environments
+        )
+
+    @staticmethod
+    def _get_release_adoption_impl(
+        now: datetime,
+        org_id: int,
+        project_releases: Sequence[Tuple[ProjectId, ReleaseName]],
+        project_ids: Sequence[ProjectId],
+        environments: Optional[Sequence[EnvironmentName]] = None,
+    ) -> ReleaseHealthBackend.ReleasesAdoption:
+        start = now - timedelta(days=1)
+
+        def _get_common_where(total: bool) -> List[Union[BooleanCondition, Condition]]:
+            where_common: List[Union[BooleanCondition, Condition]] = [
+                Condition(Column("org_id"), Op.EQ, org_id),
+                Condition(Column("project_id"), Op.IN, project_ids),
+                Condition(Column("timestamp"), Op.GTE, start),
+                Condition(Column("timestamp"), Op.LT, now),
+                Condition(
+                    Column(tag_key(org_id, "session.status")), Op.EQ, tag_value(org_id, "init")
+                ),
+            ]
+
+            if environments is not None:
+                environment_tag_values = []
+
+                for environment in environments:
+                    value = indexer.resolve(org_id, UseCase.TAG_VALUE, environment)  # type: ignore
+                    if value is not None:
+                        environment_tag_values.append(value)
+
+                where_common.append(
+                    Condition(Column(tag_key(org_id, "environment")), Op.IN, environment_tag_values)
+                )
+
+            if not total:
+                release_tag_values = []
+
+                for _, release in project_releases:
+                    value = indexer.resolve(org_id, UseCase.TAG_VALUE, release)  # type: ignore
```

Comment:
Our usage of `metric_id` and related functions is quite dangerous if there hasn't been any data ingested for that org.

For environments it's extra dangerous, and instead of filtering by an environment value that we have never seen in the tag indexer, let's skip filtering by it altogether.